### PR TITLE
Adnan mujagic/product details

### PR DESCRIPTION
**Pull request by adnan-mujagic**
Product details work in progress: 
* Added product details route at `<BASE_URL>/products/productId`
* Added an image carousel which can be swiped when the product has multiple images.
* Added some general components such as `TextHeader`
* Custom properties are now shown on the right of the product details route

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,17 +1,19 @@
...
 
 function App() {
   return (
     <div className="App">
       <Routes>
         <Route path="/" element={<Main />} />
+        <Route path="/products/:productId" element={<ProductDetails />} />
         <Route path="/login" element={<Login />} />
-        <Route path="/dashboard" element={<Main />} />
+        {/* <Route path="/dashboard" element={<Main />} /> */}
```
Comment (adnan-mujagic) [inline src/App.js:16]: Ovo sam zakomentarisao, sta ce nam na dvije rute ista stvar

File: src/components/Header/Header.jsx [hunk 2]
```diff
@@ -24,7 +24,7 @@ function Header() {
   return (
     <div className={styles.header}>
       <div className={styles["header-logo"]} onClick={() => navigate("/")}>
-        Logo
+        Quiet Depths
```
Comment (adnan-mujagic) [inline src/components/Header/Header.jsx:27]: New Title ?

File: src/components/TextHeader/TextHeader.jsx [hunk 3]
```diff
@@ -1,16 +1,28 @@
 import React from "react";
+import constants from "../../api/constants";
 
-function TextHeader({ type = "h1", text, color, textAlign, icon }) {
+function TextHeader({
+  type = "h1",
+  text,
+  color,
```
Comment (adnan-mujagic) [inline src/components/TextHeader/TextHeader.jsx:7]: Vise nacina da se customize-a ovaj `TextHader`, mozes dodadi da bude underlajnano i da ima margins etc

File: src/routes/ProductDetails/ProductDetails.jsx [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    <div className={styles["product-details"]}>
+      <Header />
+      <div className={styles["product-details-content-wrapper"]}>
+        {loading || !product.images ? (
+          <CircularProgress />
+        ) : (
+          <div className={styles["narrow"]}>
+            <div className={styles["product-content"]}>
+              <div className={styles["carousel-wrapper"]}>
+                <Carousel
```
Comment (adnan-mujagic) [inline src/routes/ProductDetails/ProductDetails.jsx:41]: Ovo je taj carousel https://www.npmjs.com/package/react-responsive-carousel

**Review by amarell — APPROVED** [review]

**Pull request merged**